tla/consensus/ccfraft.tla

```diff
@@ -859,18 +861,15 @@ AdvanceCommitIndex(i) ==
                                             LAMBDA c : c >= LastConfigurationToIndex(i, highestCommittableIndex))
               IN
               /\ configurations' = [configurations EXCEPT ![i] = new_configurations]
-              \* Retire if i is not in active configuration anymore
-              /\ IF i \notin configurations[i][Min(DOMAIN configurations'[i])]
-                 THEN \E j \in PlausibleSucessorNodes(i) :
-                    /\ LET msg == [type          |-> ProposeVoteRequest,
-                                    term          |-> currentTerm[i],
-                                    source        |-> i,
-                                    dest          |-> j ]
+            ELSE UNCHANGED <<reconfigurationVars>>
+        /\ IF membershipState'[i] = RetiredCommitted
+            THEN \E j \in PlausibleSucessorNodes(i) :
+                    /\ LET msg == [type        |-> ProposeVoteRequest,
+                                   term        |-> currentTerm[i],
+                                   source      |-> i,
+                                   dest        |-> j ]
                         IN Send(msg)
-                 \* Otherwise, states remain unchanged
-                 ELSE UNCHANGED <<messages>>
-           \* Otherwise, Configuration and states remain unchanged
-           ELSE UNCHANGED <<messages, reconfigurationVars>>
+            ELSE UNCHANGED <<messages>>
```

Comment:
Perhaps also look at this, ie. the new unchanged?

Reply:
As discussed, I think this is correct, the logic just was pulled out of the branch, but the unchanged seems correct.